libs/langchain-community/src/vectorstores/mariadb.ts

```diff
@@ -0,0 +1,760 @@
+   */
+  static async fromDocuments(
+    docs: Document[],
+    embeddings: EmbeddingsInterface,
+    dbConfig: MariaDBStoreArgs & { dimensions?: number }
+  ): Promise<MariaDBStore> {
+    const instance = await MariaDBStore.initialize(embeddings, dbConfig);
+    await instance.addDocuments(docs, { ids: dbConfig.ids });
+    return instance;
+  }
+
+  _vectorstoreType(): string {
+    return "mariadb";
+  }
+
+  /**
+   * Method to add documents to the vector store. It converts the documents into
+   * vectors, and adds them to the store.
+   *
+   * @param documents - Array of `Document` instances.
+   * @param options - Optional arguments for adding documents
+   * @returns Promise that resolves when the documents have been added.
+   */
+  async addDocuments(
+    documents: Document[],
+    options?: { ids?: string[] }
+  ): Promise<void> {
+    const texts = documents.map(({ pageContent }) => pageContent);
+
+    return this.addVectors(
+      await this.embeddings.embedDocuments(texts),
+      documents,
+      options
+    );
+  }
+
+  /**
+   * Inserts a row for the collectionName provided at initialization if it does not
+   * exist and set the collectionId.
+   */
+  private async loadCollectionId(): Promise<void> {
+    if (this.collectionId) {
+      return;
+    }
+
+    if (this.collectionTableName) {
+      const queryResult = await this.pool.query(
+        {
+          sql: `SELECT uuid from ${this.computedCollectionTableName} WHERE label = ?`,
+          rowsAsArray: true,
+        },
+        [this.collectionName]
+      );
+      if (queryResult.length > 0) {
+        this.collectionId = queryResult[0][0];
+      } else {
+        const insertString = `INSERT INTO ${this.computedCollectionTableName}(label, cmetadata) VALUES (?, ?) RETURNING uuid`;
+        const insertResult = await this.pool.query(
+          { sql: insertString, rowsAsArray: true },
+          [this.collectionName, this.collectionMetadata]
+        );
+        this.collectionId = insertResult[0][0];
+      }
+    }
+  }
+
+  /**
+   * Method to add vectors to the vector store. It converts the vectors into
+   * rows and inserts them into the database.
+   *
+   * @param vectors - Array of vectors.
+   * @param documents - Array of `Document` instances.
+   * @param options - Optional arguments for adding documents
+   * @returns Promise that resolves when the vectors have been added.
+   */
+  async addVectors(
+    vectors: number[][],
+    documents: Document[],
+    options?: { ids?: string[] }
+  ): Promise<void> {
+    const ids = options?.ids;
+
+    // Either all documents have ids or none of them do to avoid confusion.
+    if (ids !== undefined && ids.length !== vectors.length) {
+      throw new Error(
+        "The number of ids must match the number of vectors provided."
+      );
+    }
+    await this.loadCollectionId();
+
+    const insertQuery = `INSERT INTO ${this.computedTableName}(${
```

Comment:
nit: can we escape column names too/disallow nonalphanumeric?

Reply:
table name and columns names are all escaped to prevent any injection. see `escapeId` function